sbncode/CAFMaker/FillReco.cxx

```diff
@@ -88,11 +88,14 @@ namespace caf
     srshower.dEdx_plane0 = (shower.dEdx())[0];
     srshower.dEdx_plane1 = (shower.dEdx())[1];
     srshower.dEdx_plane2 = (shower.dEdx())[2];
-    srshower.energy_plane0 = (shower.Energy())[0];
-    srshower.energy_plane1 = (shower.Energy())[1];
-    srshower.energy_plane2 = (shower.Energy())[2];
+    // We need to convert the energy from MeV to GeV
+    // Also convert -999 -> -5 for consistency with other defaults in the CAFs
+    std::transform(shower.Energy().begin(), shower.Energy().end(), std::back_inserter(srshower.energy),
+        [] (float e) {return e > 0 ? e / 1000.f : -5.f;});
```

Comment:
In fact, the old code was filling `energy_planeN` and the new code is filling the `energy` vector (which I hope starts empty). I think we are fine to drop the `planeN` variables in various places, since use of the flatcafs seems to be pretty ubiquitous. But maybe that should be a separate PR?

Reply:
In practice I think this is consistent with the old behaviour, as this function is only called on an empty shower, so replacing and appending are equivalent. I am happy to add this `clear` to make this more explicit however.

Reply:
Both cases actually fill both the `energy_planeN` and `energy` vector. I agree that removing the `planeN` variants would be best in a separate PR (maybe after the upcoming production?). I guess we can discuss this in the meeting today

Reply:
I made a series of PRs for this
https://github.com/SBNSoftware/sbnanaobj/pull/23
https://github.com/SBNSoftware/sbncode/pull/159
https://github.com/SBNSoftware/sbnana/pull/41
My instinct would be to get it in before the production, but we should indeed discuss, because I don't remember all the nuances of the previous discussion.

Reply:
This would also require changes to `sbncode` to remove where we try and fill these branches right?